WordPress/WordPressUITests/Screens/BaseScreen.swift

```diff
@@ -23,13 +23,101 @@ class BaseScreen {
     }
 
     // predicate: "isEnabled == true"
-    func waitFor(predicate: String, element: XCUIElement, timeout: Int? = 5) {
-        let pred = NSPredicate(format: predicate)
-        testCase.expectation(for: pred, evaluatedWith: element, handler: nil)
-        testCase.waitForExpectations(timeout: TimeInterval(timeout!), handler: nil)
+
+    func waitFor(element: XCUIElement, predicate: String, timeout: Int? = nil) {
+        let timeoutValue = timeout ?? 5
+
+        let elementPredicate = XCTNSPredicateExpectation(predicate: NSPredicate(format: predicate), object: element)
+        _ = XCTWaiter.wait(for: [elementPredicate], timeout: TimeInterval(timeoutValue))
+    }
+
+    func waitFor2(element: XCUIElement, predicate: String,
+                                       file: String = #file, line: UInt = #line, timeout: Int? = nil) {
+        let nsPredicate = NSPredicate(format: predicate)
+        BaseScreen.testCase.expectation(for: nsPredicate,
+                    evaluatedWith: element, handler: nil)
+
+        let timeoutValue = timeout ?? 5
+
+        BaseScreen.testCase.waitForExpectations(timeout: TimeInterval(timeoutValue)) { (error) -> Void in
+            if error != nil {
+                let message = "Failed to find \(element) after \(timeoutValue) seconds."
+                BaseScreen.testCase.recordFailure(withDescription: message,
+                                   inFile: file,
+                                   atLine: line,
+                                   expected: true)
+            }
+        }
     }
 
     func isLoaded() -> Bool {
         return expectedElement.exists
     }
+
+    class func waitForLoadingIndicatorToDisappear(within timeout: TimeInterval) {
+        #if os(tvOS)
```

Comment:
Do we need this? This project is not active for tvOS.

Reply:
It was just copy-paste from `WordPressScreenshotGeneration`. Removed ✅